src/main.ts

```diff
@@ -3,6 +3,13 @@ import * as cache from './cache';
 
 export async function run() {
   try {
+    core.info('------------------------');
+    core.info('NOTE: This action is deprecated and is no longer maintained.');
+    core.info(
+      'Please, migrate to https://github.com/ruby/setup-ruby action, which is being actively maintained.'
```

Comment:
I think the `action` here is redundant, so I'd suggest just

> Please, migrate to https://github.com/ruby/setup-ruby, which is being actively maintained.

The rest reads nicely to me.

Reply:
Thanks, updated both in Readme and in logs